### ISSUE TITLE
UNIFIED-IO: A UNIFIED MODEL FOR VISION, LANGUAGE, AND MULTI-MODAL TASKS (2023)

### ISSUE THREAD

**Issue by YoojLee** [YoojLee/paper_review#38]
# Abstract
computer vision tasks, vision-and-language tasks, A large variety of AI tasks를 해결할 수 있는 UNIFIED-IO라는 모델을 제안함. 이러한 single unified model을 제안하는 데에 가장 걸림돌이 되는 것은 각 태스크 별로 input과 output이 heterogeneous하다는 것임. 우리는 각각의 input과 output을 discrete vocab token의 시퀀스로 homogenize함으로써 unification을 할 수 있었음.

이렇게 task 간 representation을 공유하게 됨으로써 하나의 transformer-based architecture를 train함으로써 vision과 language 분야에서의 90개가 넘는 다양한 데이터셋에서 joint training이 가능해짐.

UNIFIED-IO는 GRIT 벤치마크의 7개 태스크에 대해서 모두 수행할 수 있는 첫번째 모델이며, task-specific fine-tuning 없이도 16개의 다양한 벤치마크에 대해서 strong results를 produce함.

# 1. Introduction
The paper present **_UNIFIED-IO_**, the first neural model to jointly perform a large and diverse set of AI tasks spanning classical computer vision, vision-and-language, and NLP.

Unified general-purpose models avoid the need for task-specific design, learn and perform a wide range of tasks with a single architecture, can utilize large, diverse data corpora, can effectively transfer concept knowledge across tasks, and even perform tasks unknown and unobserved at design and training time.

Building unified models for computer vision has proven to be quite challenging since vision tasks have incredibly diverse input and output representations. This heterogeneity makes it very challenging to architect a single model for all these tasks. However, in NLP, inputs and desired outputs can often be uniformly represented as sequences of tokens. Thus, Seq2Seq architectures are sidely applicable to many tasks in NLP domain, and unified models employing such architectures have been central to much recent progress in NLP.

Unified models for computer vision typically use a shared visual backbone to produce visual embeddings but then employ individual branches for each of the desired tasks. (task-specific heads such as Mask R-CNN, Vision-and-Language models)

Recent works introduce modality-specific heads, for instance, a single language decoder that serves multiple tasks requiring language output like captioning and classification. However, most progress in unified models continues to be centered around V&L tasks, owing to the simplicity of building shared language decoders, and is often limited to supporing just a handful of tasks.

Unified-IO is a Seq2Seq model capable of performing a variety of tasks using a unified architecture without a need for either task or even modality-specific branches by homogenizing every task's output into a sequence of discrete tokens. 
→ Then, how they did tokenization?

1) Dense structured outputs (e.g., images, segmentation masks, and depth maps)
: converted to sequences using a vector quantization variational auto-encoder (VQ-VAE)

2) sparse structured outputs (bounding boxes, human joint locations)
: transcribed into sequences of coordinate tokens

3) language outputs
: converted to sequences using byte-pair encoding.

## Contribution
- the first model to support all 7 tasks in the General Robust Image Task (GRIT) Benchmark
- evaluated on 16 diverse benchmarks across computer vision and NLP, without any fine-tuning towards any individual benchmark, and find that it performs remarkably well than specialized SOTA models.

# 2. Vision, Language, and Multi-modal tasks
Unified-IO is designed to handle a wide range of language, vision and language, and classic vision tasks in a unified way. These datasets cover a wide range of tasks, skills, and modalities.

categorize the input and output modalities of each task into 4 different types: Text. Image, Sparse (a small number of location coordinates within the image), Dense (per-pixel labels such as depth maps, surface normal maps, etc.)

Grouped datasets into 8 groups and 22 tasks.
